resolwe_bio/processes/support_processors/seqtk.py

```diff
@@ -0,0 +1,324 @@
+
+from resolwe.process import (
+    BooleanField,
+    Cmd,
+    DataField,
+    FileField,
+    FileHtmlField,
+    FloatField,
+    GroupField,
+    IntegerField,
+    ListField,
+    Persistence,
+    Process,
+    SchedulingClass,
+)
+
+
+class SeqtkSampleSingle(Process):
+    """Subsample reads from FASTQ file (single-end).
+
+    [Seqtk](https://github.com/lh3/seqtk) is a fast and lightweight tool for
+    processing sequences in the FASTA or FASTQ format. The Seqtk "sample" command
+    enables subsampling of the large FASTQ file(s).
+    """
+
+    slug = "seqtk-sample-single"
+    name = "Subsample FASTQ (single-end)"
+    process_type = "data:reads:fastq:single:seqtk"
+    version = "1.3.0"
+    category = "Other"
+    data_name = '{{ reads|sample_name|default("?") }}'
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.CACHED
+    entity = {
+        "type": "sample",
+    }
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/s4q6j6e8/resolwebio/common:2.3.1"}
+        },
+        "resources": {
+            "cores": 1,
+            "memory": 16384,
+        },
+    }
+
+    class Input:
+        """Input fields to process SeqtkSampleSingle."""
+
+        reads = DataField(
+            "reads:fastq:single",
+            label="Reads",
+        )
+        n_reads = IntegerField(
+            label="Number of reads",
+            default=1000000,
+        )
+
+        class Advanced:
+            """Advanced options."""
+
+            seed = IntegerField(
+                label="Seed",
+                default=11,
+            )
+            fraction = FloatField(
+                label="Fraction",
+                required=False,
+                description="Use the fraction of reads [0 - 1.0] from the "
+                "original input file instead of the absolute number of reads. "
+                "If set, this will override the 'Number of reads' input parameter.",
+            )
+            two_pass = BooleanField(
+                label="2-pass mode",
+                default=False,
+                description="Enable two-pass mode when down-sampling. "
+                "Two-pass mode is twice as slow but with much reduced memory.",
+            )
+
+        advanced = GroupField(
+            Advanced,
+            label="Advanced options",
+        )
+
+    class Output:
+        """Output fields to process SeqtkSampleSingle."""
+
+        fastq = ListField(
+            FileField(),
+            label="Remaining reads",
+        )
+        fastqc_url = ListField(
+            FileHtmlField(),
+            label="Quality control with FastQC",
+        )
+        fastqc_archive = ListField(
+            FileField(),
+            label="Download FastQC archive",
+        )
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+
+        if inputs.advanced.fraction and not 0 < inputs.advanced.fraction < 1.0:
```

Comment:
Please check if this should be `0 < inputs.advanced.fraction <= 1` instead

Reply:
FloatFields also support the range attribute that could replace this if sentence. 
```
fraction = FloatField(
    label="Fraction",
    required=False,
    range=[0.0, 1.0],
    description=...,
)
```